pkg/readiness/object_tracker.go

```diff
@@ -56,15 +60,20 @@ type objectTracker struct {
 	populated     bool                      // all expectations have been provided
 	allSatisfied  bool                      // true once all expectations have been satisfied. Acts as a circuit-breaker.
 	kindsSnapshot []schema.GroupVersionKind // Snapshot of kinds before freeing memory in Satisfied.
+	mutators      []objDataMutator          // functions that mutate objData types during their creation.  Allows for defaults.
 }
 
-func newObjTracker(gvk schema.GroupVersionKind) *objectTracker {
+func newObjTracker(gvk schema.GroupVersionKind, fns ...objDataMutator) *objectTracker {
```

Comment:
Golang only lets you have one variadic argument type, I'm not sure we want to spend it exclusively on configuring object data specifically.

We could just have:

```
func newObjTracker(gvk schema.GroupVersionKind, mutator objDataMutator) *objectTracker {
   if mutator == nil {
      mutator = setRetriesFromFlag
   }
}
```

Also, if we are mutating anyway, why not just pass a constructor method that creates an appropriately formed object? That would save the need to pass arguments to `mutatedObjData()` on line 115 of the new code (though the name would likely change).

Reply:
I did my way because I wouldn't have to change the usages of `newObjTracker` around the code, though admittedly that isn't much of a reason.

I've changed it to what you have above.

Given that there isn't the ability to pass in arbitrary `objDataMutator`s, I went ahead and changed `objectTracker.mutators` (of type `[]objDataMutator`) to a singular `objectTracker.mutator` (type `objDataMutator`).  LMK if you think I should switch it back.

Reply:
Are [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) still cool these days? That's what I usually use. Fine either way.

Reply:
I used a functional options pattern initially, and @maxsmythe advocated for saving the variadic parameter for the future if possible.  I figure this method leaves things open-ended for the future and perhaps follows a "YAGNI" mentality.